android/src/test/java/org/mozilla/mozstumbler/client/UpdaterTest.java

```diff
@@ -90,9 +92,11 @@ public void testUpdaterThrottlesRequests() {
         ServiceLocator.getInstance().putService(IHttpUtil.class, mockHttp);
         ServiceLocator.getInstance().putService(ISystemClock.class, clock);
 
-        // wifi is always unavailable
-        doReturn(false).when(upd).wifiExclusiveAndUnavailable(Mockito.any(Context.class));
+        ConnectivityManager connectivityManager = (ConnectivityManager) ctx.getSystemService(Context.CONNECTIVITY_SERVICE);
+        MyShadowConnectivityManager shadowConnManager = (MyShadowConnectivityManager) Robolectric.shadowOf(connectivityManager);
 
+        shadowConnManager.setConnectedFlag(true);
+        doReturn(false).when(upd).wifiExclusiveAndUnavailable(Mockito.any(Context.class));
```

Comment:
You are right, this does not seem to be necessary.
When I ran my modified tests this failed because there was no network connection, so I added `shadowConnManager.setConnectedFlag(true)`. But I could not reproduce this problem, so I reverted these changes now.